### PR TITLE
Add test for #11436 (missing StreamerInfo for nested non-split memberwise streamed collection)

### PR DESCRIPTION
**Pull request by pcanal**
Also re-enabled 2 test directories that were disabled in the port to cmake/ctest and removal of Makefile.

### CODE REVIEW

File: root/io/TFile/execNestedCollectionStreamerInfo.cxx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  std::vector<Inside> fInside;
+};
+
+struct Outer {
+  std::vector<Middle> fMiddle;
+};
+
+bool CheckNestedCollectionStreamerInfo(bool empty = true, const char *filename = "forceinfo.root")
+{
+   std::unique_ptr<TFile> f{ new TMemFile(filename, "RECREATE") };
```
Comment (Axel-Naumann) [inline root/io/TFile/execNestedCollectionStreamerInfo.cxx:22]: I prefer tests with as little complexity as possible :-)

```suggestion
bool CheckNestedCollectionStreamerInfo(bool empty = true)
{
   std::unique_ptr<TFile> f{ new TMemFile("forceinfo.root", "RECREATE") };
```

File: root/io/TFile/execNestedCollectionStreamerInfo.cxx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	  << (empty ? "empty" : "with content") << "\n";
+     return false;
+   }
+   return true;
+}
+
+int execNestedCollectionStreamerInfo()
+{
+   bool result = CheckNestedCollectionStreamerInfo(true) && CheckNestedCollectionStreamerInfo(false);
+   return result ? 0 : 1;
```
Comment (Axel-Naumann) [inline root/io/TFile/execNestedCollectionStreamerInfo.cxx:48]:
```suggestion
   bool success = CheckNestedCollectionStreamerInfo(true) && CheckNestedCollectionStreamerInfo(false);
   return success ? 0 : 1;
```

File: root/meta/genreflex/ROOT-5881/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+                  FIXTURES_SETUP execStringInputFiles)
+
+ROOTTEST_ADD_TEST(execstringI
+  MACRO execstringI.C
+  OUTREF execstringI.ref
+  DEPENDS execStringInputFiles
+  FIXTURES_REQUIRED execStringInputFiles
+)
+
+#CMakeFiles
```
Comment (Axel-Naumann) [inline root/meta/genreflex/ROOT-5881/CMakeLists.txt:24]:
```suggestion
```

File: root/io/TFile/execNestedCollectionStreamerInfo.cxx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+#include <vector>
+#include "TTree.h"
+#include "TFile.h"
+#include "TMemFile.h"
+#include <iostream>
+#include <memory>
+
+struct Inside {
```
Comment (Axel-Naumann) [inline root/io/TFile/execNestedCollectionStreamerInfo.cxx:8]: Could you add a comment why we don't need a dictionary for any of these classes?

File: root/io/TFile/execNestedCollectionStreamerInfo.cxx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+struct Middle {
+  std::vector<Inside> fInside;
+};
+
+struct Outer {
+  std::vector<Middle> fMiddle;
+};
+
+bool CheckNestedCollectionStreamerInfo(bool empty = true, const char *filename = "forceinfo.root")
```
Comment (Axel-Naumann) [inline root/io/TFile/execNestedCollectionStreamerInfo.cxx:20]: Could you add a comment why `empty` and `!empty` might give different results? And how this tests #11436 ?

**Review by Axel-Naumann — APPROVED** [review]
But see comments

File: root/io/TFile/execNestedCollectionStreamerInfo.cxx (hunk 5 above)
Comment (pcanal) [reply root/io/TFile/execNestedCollectionStreamerInfo.cxx:20]:
```
   // Test both empty and filled collections as the StreamerInfo handling is different
   // in some cases (split vs non-split, object-wise streaming vs member-wise streaming).
   // In the case of the original issue (https://github.com/root-project/root/issues/11436)
   // having a filled collection did not help.
```

**Comment by pcanal** [comment]
@phsft-bot build

**Comment by pcanal** [comment]
@phsft-bot build just on ROOT-performance-centos8-multicore/cxx17

**Comment by pcanal** [comment]
@phsft-bot build just on mac1015/cxx17

**Pull request merged**